src/core/linux/SDL_udev.c

```diff
@@ -365,7 +365,7 @@ static void get_caps(struct udev_device *dev, struct udev_device *pdev, const ch
     }
 }
 
-static int guess_device_class(struct udev_device *dev)
+static SDL_UDEV_deviceclass guess_device_class(struct udev_device *dev)
```

Comment:
I think SDL intentionally uses a plain integer in situations where the semantics intended are "bitwise-or of zero or more flags chosen from enum SDL_UDEV_deviceclass", because a sufficiently pedantic compiler would be within its rights to say that, while `SDL_UDEV_DEVICE_KEYBOARD` and `SDL_UDEV_DEVICE_MOUSE` are valid members of `SDL_UDEV_deviceclass`, strictly speaking `SDL_UDEV_DEVICE_KEYBOARD|SDL_UDEV_DEVICE_MOUSE` is not.

However, bitwise-or'ing flags like that is ubiquitous in ecosystems like GLib/GTK and Qt, so we know that in practice it works well.

I'll defer to Sam and other SDL maintainers on what is preferred here.

Reply:
I had put it in because I thought it told you what values the `int` could have. I can see the argument that math doesn't technically work on enums though. I pulled this stuff out to a commit that sits on top instead of on bottom. So it can be taken or left.